### PR TITLE
check that requested cores >= 1 and < node core total

### LINKED ISSUE #92: Validate # of core per command
Tell the user that his job can't be launched on this queue if one command requires more cores than 1 node can provide.

Let say one node of a cluster has 24 cores, if a user set --coresPerCommand 30 it should alert him and not launch the jos.

We should also verify that it cannot be set lower than 1.

### PR DESCRIPTION
**Pull request by slefrancois**
Fixes #92

### CODE REVIEW

File: scripts/smart-dispatch [hunk 1]
```diff
@@ -113,6 +113,12 @@ def main():
     # TODO: use args.memPerNode instead of args.memPerNode
     queue = Queue(args.queueName, CLUSTER_NAME, args.walltime, args.coresPerNode, args.gpusPerNode, float('inf'), args.modules)
 
+    # Check that requested core number does not exceed node total
+    if args.coresPerCommand > queue.nb_cores_per_node:
+        sys.stderr.write("smart-dispatch: error: coresPerCommand exceeds nodes total: asked {req_cores} cores, nodes have {node_cores}\n"
+                         .format(req_cores=args.coresPerCommand, node_cores=queue.nb_cores_per_node))
+        sys.exit(2)
```
Comment (MarcCote) [inline scripts/smart-dispatch]: Is there any particular reason for `sys.exit(2)` instead of `sys.exit(1)` ?

File: scripts/smart-dispatch [hunk 2]
```diff
@@ -175,6 +181,8 @@ def parse_arguments():
             parser.error("You need to specify a command to launch.")
         if args.queueName not in AVAILABLE_QUEUES and ((args.coresPerNode is None and args.gpusPerNode is None) or args.walltime is None):
             parser.error("Unknown queue, --coresPerNode/--gpusPerNode and --walltime must be set.")
+        if args.coresPerCommand < 1:
+            parser.error("coresPerNode must be at least 1")
```
Comment (MarcCote) [inline scripts/smart-dispatch]: Good catch.

**Comment by MarcCote** [comment]
Can you make unit tests to check your fix?

File: scripts/smart-dispatch (hunk 1 above)
Comment (slefrancois) [inline scripts/smart-dispatch]: 2 is usually the code for a command line error, it's what parser.error() returns.

Comment (MarcCote) [inline scripts/smart-dispatch]: Cool, didn't know that.

**Comment by mgermain** [comment]
Yup just add some tests in `test_smart_dispatch.py` and I'm good with merging this.

**Comment by MarcCote** [comment]
LGTM

**Pull request merged**